### PR TITLE
Feature/cypress testing

### LINKED ISSUE #18: Cypress testing

### PR DESCRIPTION
**Pull request by Baskanbetul**
What does this PR do?
- Completed testing and added mock data for zip and trees

Where should your reviewer start?
- End to end directory
- Line 77 on App.js
- Line 48 on DetailsContainer.js
- Line 5 on Error.js
- Line 7 on Home.js

What (if any) features are you implementing?
- Add testing for returning from home page to home page 

What (if anything) did you refactor?
- Header component 
- DetailsContainer component
- Error component
- App component

Were there any issues that arose?
- Could not test outgoing link to Nature Hills

Is there anything that you need from your teammate?
- N/A

Any other comments, questions, or concerns?
- N/A

Resolves #18

### CODE REVIEW

File: cypress/e2e/DetailsContainer.cy.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  beforeEach(() => {
+    cy.intercept('GET', 'https://tree-pal-api.herokuapp.com/api/v1/trees', {fixture : 'getTreeMockData.json'});
+    cy.intercept('GET', 'https://tree-pal-api.herokuapp.com/api/v1/zip', {fixture : 'getZipMockData.json'});
+    cy.visit('http://localhost:3000/');
+    cy.get('[data-cy="header"]').contains('Treezy');
+    cy.get('[data-cy="zip-code-entry"]').type('80101');
+    cy.get('[data-cy="go-button"]').invoke('attr', 'value').should('contain', 'GO!')
+    cy.get('[data-cy="go-button"]').click()
+    cy.url().should('eq', 'http://localhost:3000/result')
+    cy.get('.trees-container').within(() => {
```
Comment (BrandonAinsworth) [inline cypress/e2e/DetailsContainer.cy.js:11]: Nice use of cypress within, very cool! 🧪

File: cypress/e2e/DetailsContainer.cy.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  it('Should have leaf button', () => {
+    cy.get('[data-cy="leaf-button"]').should('have.attr','value','LEAF!').click()
+  })
+
+  it('Should be able to go back to the home page when clicking logo', () => {
+    cy.get('[data-cy="header"]').click()
+    cy.url().should('eq', 'http://localhost:3000/');
+  })
+
+  it('Should be able to go ', () => {
```
Comment (BrandonAinsworth) [inline cypress/e2e/DetailsContainer.cy.js:41]: A toads gotta do what a toads gotta do. 💪

File: cypress/e2e/Home.cy.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    cy.visit('http://localhost:3000/');
+  })
+
+  it('The user should see the home page', () => {
+    cy.get('[data-cy="header"]').contains('Treezy');
+    cy.get('[data-cy="zip-code-entry"]').invoke('attr', 'placeholder').should('contain', 'Enter 5-Digit CO Zip Code');
+    cy.get('[data-cy="go-button"]').invoke('attr', 'value').should('contain', 'GO!');
+  })
+
+  it('Should be able to go back to the home page when clicking logo', () => {
```
Comment (BrandonAinsworth) [inline cypress/e2e/Home.cy.js:14]: Glad we caught this, close one! 😅

**Review by BrandonAinsworth — APPROVED** [review]

File: cypress/e2e/DetailsContainer.cy.js (hunk 1 above)
Comment (irmakerdem) [inline cypress/e2e/DetailsContainer.cy.js:11]: Cool way to test using a `within()`.

File: cypress/e2e/DetailsContainer.cy.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  })
+
+  it('Should display about the tree', () => {
+    cy.get('.details-middle').contains('About This Tree')
+    cy.get('.bullet-list').should('have.length', 1).contains('Attractive Foliage')
+  })
+
+  it('Should have attributes', () => {
+    cy.get('.more-about').contains('Attributes')
+    cy.get('.emojis').should('have.length', 9).contains('🐝')
```
Comment (irmakerdem) [inline cypress/e2e/DetailsContainer.cy.js:29]: 🐝

File: src/components/Error/Error.js [hunk 5]
```diff
@@ -2,17 +2,17 @@ import React from 'react';
...
+	return (
+		<>
+			<Header clearZipTrees={clearZipTrees} />
+			<div className="error-messaging">
+				<h4>Thank you for visiting our website.</h4>
+				<p>You've found a page that we cannot find.</p>
+				<p>Please return to our 🏡 home page using the Treezy link above.</p>
+			</div>
+		</>
+	);
```
Comment (rachel-bock) [inline src/components/Error/Error.js:15]: There seems to be some indentation differences in the Error component.  Did we want the indentation like this?

File: cypress/e2e/Error.cy.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+describe('Home', () => {
+  beforeEach(() => {
+    cy.intercept('GET', 'https://tree-pal-api.herokuapp.com/api/v1/trees', {fixture : 'getTreeMockData.json'});
+    cy.intercept('GET', 'https://tree-pal-api.herokuapp.com/api/v1/zip', {fixture : 'getZipMockData.json'});
+    cy.visit('http://localhost:3000/banana/');
```
Comment (irmakerdem) [inline cypress/e2e/Error.cy.js:5]: 🍌

**Review by rachel-bock — APPROVED** [review]
🌳 Yeah!  Passing tests are awesome!

File: cypress/e2e/SearchResult.cy.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    cy.get('[data-cy="go-button"]').invoke('attr', 'value').should('contain', 'GO!')
+    cy.get('[data-cy="go-button"]').click()
+    cy.url().should('eq', 'http://localhost:3000/result')
+  })
+
+  it('The user should see search results', () => {
+    cy.get('.results-top').contains('AGATE')
+    cy.get('.tree-name').contains('Jane Magnolia')
+    cy.get('.trees-container').within(() => {
+      cy.get('.tree-image').should('have.length', 1);
```
Comment (irmakerdem) [inline cypress/e2e/SearchResult.cy.js:17]: Length of one because there is one ul! 😱

**Review by irmakerdem — APPROVED** [review]
Yay for getting testing done! 🧪

**Pull request merged**